pkg/api/api.go

```diff
@@ -294,14 +295,17 @@ func (a *API) RegisterIngester(i Ingester, pushConfig distributor.Config) {
 	a.indexPage.AddLink(SectionDangerous, "/ingester/flush", "Trigger a Flush of data from Ingester to storage")
 	a.indexPage.AddLink(SectionDangerous, "/ingester/shutdown", "Trigger Ingester Shutdown (Dangerous)")
 	a.indexPage.AddLink(SectionDangerous, "/ingester/renewTokens", "Renew Ingester Tokens (10%)")
+	a.indexPage.AddLink(SectionDangerous, "/ingester/mode", "Change Ingester mode on ring")
 	a.RegisterRoute("/ingester/flush", http.HandlerFunc(i.FlushHandler), false, "GET", "POST")
 	a.RegisterRoute("/ingester/shutdown", http.HandlerFunc(i.ShutdownHandler), false, "GET", "POST")
 	a.RegisterRoute("/ingester/renewTokens", http.HandlerFunc(i.RenewTokenHandler), false, "GET", "POST")
+	a.RegisterRoute("/ingester/mode", http.HandlerFunc(i.ModeHandler), false, "GET", "POST")
 	a.RegisterRoute("/ingester/push", push.Handler(pushConfig.MaxRecvMsgSize, a.sourceIPs, i.Push), true, "POST") // For testing and debugging.
 
 	// Legacy Routes
 	a.RegisterRoute("/flush", http.HandlerFunc(i.FlushHandler), false, "GET", "POST")
 	a.RegisterRoute("/shutdown", http.HandlerFunc(i.ShutdownHandler), false, "GET", "POST")
+	a.RegisterRoute("/mode", http.HandlerFunc(i.ModeHandler), false, "GET", "POST")
```

Comment:
Is "GET" a supported operation? The implementation and test seems to only support "POST".

Reply:
It does support "GET". I added some tests for it, but this is slight weird for me. I would not expect a "GET" to make changes on the service, but checking all others APIs are "GET" and "POST. I think we do that to make it easier to accept the requests from index pages. I would consider dropping "GET" to the new API, but we would have to change how the index works to accept "POST", right now it seems that it just redirect to the desired path

Reply:
people calling this from index page need GET, otherwise we have to do something weird with a form. I prefer like it is now, it's simpler.